examples/graphmask_explainer.py

```diff
@@ -0,0 +1,337 @@
+                                   layer_type='GAT')
+    feat_mask = explainer.train_graph_explainer(4, dataset[4].x,
+                                                dataset[4].edge_index)
+    edge_mask = explainer.explain_graph(4)
+    ax, G = explainer.visualize_subgraph(None, dataset[4].edge_index,
+                                         edge_mask, y=dataset[4].y,
+                                         edge_y=edge_y, node_alpha=feat_mask,
+                                         task='graph')
+    plt.show()
+
+    # RGCN Graph Classification===============================================
+    dataset = TUDataset(root='/tmp/ENZYMES', name='ENZYMES')
+    loader = DataLoader(dataset, batch_size=100, shuffle=True)
+    edge_type = torch.randint(low=0, high=90,
+                              size=(dataset[4].edge_index.size(1), ))
+    edge_y = torch.randint(low=0, high=30,
+                           size=(dataset[4].edge_index.size(1), ))
+
+    class Net(torch.nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.conv1 = FastRGCNConv(3, 6, 90, num_blocks=3, aggr='mean')
+            self.conv2 = FastRGCNConv(6, 3, 90, num_blocks=3, aggr='add')
+            self.conv3 = FastRGCNConv(3, 6, 90, num_blocks=3, aggr='mean')
+
+        def forward(self, x, edge_index, batch, edge_type):
+            x = F.relu(self.conv1(x, edge_index, edge_type))
+            x = F.dropout(x, training=self.training)
+            x = F.relu(self.conv2(x, edge_index, edge_type))
+            x = F.dropout(x, training=self.training)
+            x = F.relu(self.conv3(x, edge_index, edge_type))
+            x = global_mean_pool(x, batch)
+            return F.log_softmax(x, dim=1)
+
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    model = Net().to(device)
+    optimizer = torch.optim.Adam(model.parameters(), lr=0.01,
+                                 weight_decay=5e-4)
+    explainer = GraphMaskExplainer(3, epochs=4)
+    explainer.__set_flags__(model)
+
+    for epoch in range(1, 5):
+        loss_all = 0
+        for data in loader:
+            model.train()
+            optimizer.zero_grad()
+            data.edge_type = torch.randint(low=0, high=90,
+                                           size=(data.edge_index.size(1), ))
+            output = model(data.x, data.edge_index, data.batch, data.edge_type)
+            loss = F.nll_loss(output, data.y)
+            loss.backward()
+            torch.nn.utils.clip_grad_value_(model.parameters(), 2.0)
+            optimizer.step()
+            loss_all += loss.item() * data.num_graphs
+        print('Epoch {:03d}'.format(epoch),
+              'train_loss: {:.4f}'.format(loss_all / len(loader.dataset)))
+    print("Optimization Finished!")
+
+    explainer = GraphMaskExplainer(3, model_to_explain=model, epochs=4,
+                                   layer_type='FastRGCN')
+    feat_mask = explainer.train_graph_explainer(4, dataset[4].x,
```

Comment:
Why is `train_graph_explainer` and `explain_graph` separated into two steps?

Reply:
I think it does make sense to have one function to only train the explainer and another one dedicated to only make explanations. One point to note here is that `train_graph_explainer` has `for` loops to train the explainer, just making attempts to integrating bodies of both functions (`train_graph_explainer` and `explain_graph`) will surely make the whole interface look so vague and overly skeptical. WDYT??